apps/emqx_management/test/emqx_mgmt_api_clients_SUITE.erl

```diff
@@ -171,6 +200,257 @@ t_clients(_) ->
+    APIPort = 18084,
+    Port1 = get_mqtt_port(N1, tcp),
+    Port2 = get_mqtt_port(N2, tcp),
+    Connect = fun(Opts) ->
+        Defaults = #{
+            expiry => 30,
+            clean_start => false
+        },
+        #{
+            port := Port,
+            clientid := ClientId,
+            clean_start := CleanStart,
+            expiry := EI
+        } = maps:merge(Defaults, Opts),
+        {ok, C} = emqtt:start_link([
+            {port, Port},
+            {proto_ver, v5},
+            {clientid, ClientId},
+            {clean_start, CleanStart},
+            {properties, #{'Session-Expiry-Interval' => EI}}
+        ]),
+        {ok, _} = emqtt:connect(C),
+        C
+    end,
+    AssertSingleClient = fun(Node, ClientId, Connected) ->
+        IsConnected =
+            case Connected of
+                connected -> true;
+                disconnected -> false
+            end,
+        ?retry(
+            100,
+            20,
+            ?assertMatch(
+                {ok, {{_, 200, _}, _, #{<<"data">> := [#{<<"connected">> := IsConnected}]}}},
+                list_request(APIPort)
+            )
+        ),
+        ?retry(
+            100,
+            20,
+            ?assertMatch(
+                {ok, {{_, 200, _}, _, #{<<"data">> := [#{<<"connected">> := IsConnected}]}}},
+                list_request(APIPort, "node=" ++ atom_to_list(Node)),
+                #{node => Node}
+            )
+        ),
+        ?assertMatch(
+            {ok, {{_, 200, _}, _, #{<<"connected">> := IsConnected}}},
+            lookup_request(ClientId, APIPort)
+        ),
+        ok
+    end,
+
+    ?assertMatch({ok, {{_, 200, _}, _, #{<<"data">> := []}}}, list_request(APIPort)),
+
+    ?check_trace(
+        begin
+            %% Scenario 1
+            %% 1) Client connects and is listed as connected.
+            %% 2) Client disconnects and is listed as disconnected.
+            %% 3) Client reconnects and is listed as connected.
+            %% 4) Client disconnects.
+            %% 5) Session is GC'ed, client is removed from list.
+            ?tp(notice, "scenario 1", #{}),
+            ClientId1 = <<"c1">>,
+            C1 = Connect(#{port => Port1, clientid => ClientId1}),
+            AssertSingleClient(N1, ClientId1, connected),
+            ok = emqtt:disconnect(C1),
+            AssertSingleClient(N1, ClientId1, disconnected),
+            C2 = Connect(#{port => Port1, clientid => ClientId1}),
+            AssertSingleClient(N1, ClientId1, connected),
+            ok = emqtt:stop(C2),
+            %% simulate GC
+            ?tp(notice, "gc", #{}),
+            ok = erpc:call(N1, emqx_persistent_session_ds, destroy_session, [ClientId1]),
+            ?retry(
+                100,
+                20,
+                ?assertMatch(
+                    {ok, {{_, 200, _}, _, #{<<"data">> := []}}},
+                    list_request(APIPort)
+                )
+            ),
+
+            %% Scenario 2
```

Comment:
Nit. Given these are different test scenarios, it feels odd to squeeze them into a single testcase. Having 5 separate testcases should make possible future regressions more isolated and easier to analyze.